### PR TITLE
tracing: add upstream address to tracing child spans

### LINKED ISSUE #8954: Add upstream address to tracing child spans
Would help to know to which upstream endpoint the request went to.

### PR DESCRIPTION
**Pull request by ascheglov**
Would help to know to which upstream endpoint the request went to.

Risk Level: Low
Testing: Updated existing tests
Docs Changes: yes, updated "What data each trace contains"
Release Notes: yes
Fixes #8954

### CODE REVIEW

**Comment by ascheglov** [comment]
I've updated the docs.

> A question though, do you think sending this IP address to be collected by tracing collector will be fine for all users?

We already know all the endpoints since its in the config, we can write upstream IP into access logs with `%UPSTREAM_HOST%`, so I doubt that traces could compromise much.

However it might make sense to make the tags set configurable. We have like less than 64 tags, so a quick lookup over a tag mask is totally feasible.

File: source/common/tracing/http_tracer_impl.cc [hunk 1]
```diff
@@ -187,6 +187,11 @@ void HttpTracerUtility::finalizeUpstreamSpan(Span& span, const Http::HeaderMap*
   span.setTag(Tracing::Tags::get().HttpProtocol,
               AccessLog::AccessLogFormatUtils::protocolToString(stream_info.protocol()));
 
+  if (stream_info.upstreamHost() && stream_info.upstreamHost()->address()) {
```
Comment (mattklein123) [inline source/common/tracing/http_tracer_impl.cc]: The upstream host must have an address so I don't think the second part of this if statement is needed.

File: docs/root/intro/version_history.rst [hunk 2]
```diff
@@ -19,6 +19,7 @@ Version history
 * tcp_proxy: added :ref:`ClusterWeight.metadata_match<envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.WeightedCluster.ClusterWeight.metadata_match>`
 * tcp_proxy: added :ref:`hash_policy<envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.hash_policy>`
 * thrift_proxy: added support for cluster header based routing.
+* tracing: added upstream_address tag.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: nit: alpha order (can you fix the others as well)

File: source/common/tracing/http_tracer_impl.cc (hunk 1 above)
Comment (ascheglov) [reply source/common/tracing/http_tracer_impl.cc]: yep, removed

File: docs/root/intro/version_history.rst (hunk 2 above)
Comment (ascheglov) [reply docs/root/intro/version_history.rst]: sorted

**Pull request merged**